src/core/emergency-action.c

```diff
@@ -35,47 +35,47 @@ static void log_and_status(Manager *m, const char *message) {
                               "%s", message);
 }
 
-int failure_action(
+int emergency_action(
                 Manager *m,
-                FailureAction action,
+                EmergencyAction action,
                 const char *reboot_arg) {
 
         assert(m);
         assert(action >= 0);
-        assert(action < _FAILURE_ACTION_MAX);
+        assert(action < _EMERGENCY_ACTION_MAX);
 
-        if (action == FAILURE_ACTION_NONE)
+        if (action == EMERGENCY_ACTION_NONE)
                 return -ECANCELED;
 
         if (!MANAGER_IS_SYSTEM(m)) {
                 /* Downgrade all options to simply exiting if we run
                  * in user mode */
 
-                log_warning("Exiting as result of failure.");
+                log_warning("Exiting as result of emergency.");
```

Comment:
That's not a good change for the "failure" case. Maybe `emergency_action` should take a parameter (const char* ?), that could be used to customize those messages depending whether this point is reached from a user controlled action or failure handling.